### ISSUE TITLE
Remove the 4-byte alignment requirement for the JIT allocator: ArenaAllocator::allocateMemory

### ISSUE THREAD

**Issue by briansull** [dotnet/coreclr#13093]
```
    assert(size != 0 && (size & (sizeof(int) - 1)) == 0);

    // Ensure that we always allocate in pointer sized increments.
    size = (size_t)roundUp(size, sizeof(size_t));
```

It is unnecessary to assert that we have a four byte aligned request, since we immediately roundUp to a pointer sized request size.

**Comment by briansull** [comment]
@pgavlin FYI

**Comment by pgavlin** [comment]
Thanks @briansull.

I think it's actually `Compiler::Alloc` and its pair, `ArrayAlloc` that need to do the alignment, as they may be called from functions besides these operators.

cc @dotnet/jit-contrib

**Comment by RussKeldorph** [comment]
@briansull Should this have been closed?

**Comment by karelz** [comment]
I guess it is tracking the 2.0.x port now ...

**Comment by briansull** [comment]
This can be closed as the fix was ported to 2.0 with PR #13412
